src/main/java/org/folio/inventory/resources/AbstractInstances.java

```diff
@@ -56,17 +57,25 @@ public AbstractInstances(final Storage storage, final HttpClient client) {
     this.config = new InventoryConfigurationImpl();
   }
 
+  protected CompletableFuture<List<Response>> updateRelatedRecords(
+    RoutingContext routingContext, WebContext context, Instance instance) {
+
+    return updateRelatedInstances(instance, routingContext, context)
+      .thenCompose(r -> updateInstanceRelationships(instance, routingContext, context))
+      .thenCompose(r -> updatePrecedingSucceedingTitles(instance, routingContext, context));
+  }
```

Comment:
Thank you for extracting this out, the future chains were getting a little unwieldy.